File: app/src/main/java/vandyke/siamobile/backend/BootReceiver.java
/*
 * Copyright (c) 2017 Nicholas van Dyke
 *
 * This file is subject to the terms and conditions defined in Licensing section of the file 'README.md'
 * included in this source code package. All rights are reserved, with the exception of what is specified there.
 */

package vandyke.siamobile.backend;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import vandyke.siamobile.SiaMobileApplication;
import vandyke.siamobile.backend.coldstorage.ColdStorageService;
import vandyke.siamobile.backend.siad.SiadMonitorService;
import vandyke.siamobile.backend.wallet.WalletMonitorService;

public class BootReceiver extends BroadcastReceiver {
    public void onReceive(Context context, Intent intent) {

        if (SiaMobileApplication.prefs.getBoolean("runInBackground", false)) {
            if (SiaMobileApplication.prefs.getString("operationMode", "cold_storage").equals("cold_storage"))
                context.startService(new Intent(context, ColdStorageService.class));
            else if (SiaMobileApplication.prefs.getString("operationMode", "cold_storage").equals("local_full_node"))
                context.startService(new Intent(context, SiadMonitorService.class));
            context.startService(new Intent(context, WalletMonitorService.class));
        }
    }
}

File: app/src/main/java/vandyke/siamobile/backend/BootReceiver.kt
/*
 * Copyright (c) 2017 Nicholas van Dyke
 *
 * This file is subject to the terms and conditions defined in 'LICENSE.md'
 */

package vandyke.siamobile.backend

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import vandyke.siamobile.backend.coldstorage.ColdStorageService
import vandyke.siamobile.backend.siad.SiadService
import vandyke.siamobile.backend.wallet.WalletService
import vandyke.siamobile.prefs

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        /* need to check for Android O in some places because Android O will cause a crash when attempting to start a
        * background service when the app is not in foreground */
        if (prefs.runInBackground) {
            when (prefs.operationMode) {
                "cold_storage" -> {
                    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
                        context.startService(Intent(context, ColdStorageService::class.java))
                }
                "local_full_node" -> context.startService(Intent(context, SiadService::class.java))
            }
            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O)
                context.startService(Intent(context, WalletService::class.java))
        }
    }
}
